clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java

```diff
@@ -88,22 +85,7 @@
   public FSDataOutputStream append(Path cPath, int bufferSize, Progressable progress)
       throws IOException {
     LOG.info("append({}, {}, {})", cPath, bufferSize, progress);
-    if (mStatistics != null) {
-      mStatistics.incrementWriteOps(1);
-    }
-    TachyonURI path = new TachyonURI(Utils.getPathWithoutScheme(cPath));
-    TachyonFile file;
-    try {
-      file = mTFS.open(path);
-      if (mTFS.getInfo(file).length > 0) {
-        LOG.warn("Appending to nonempty file. This may be an error.");
-      }
-    } catch (TachyonException e) {
-      throw new IOException(e);
-    }
-
-    return new FSDataOutputStream(mTFS.getOutStream(file.getFileId(), OutStreamOptions.defaults()),
-        mStatistics);
+    throw new UnsupportedOperationException("Append is not supported in Tachyon.");
```

Comment:
Good point, I don't think the previous logic actually works since IOException will be thrown always, but I've reverted it to have the behavior you mentioned. Though I'm not sure if that's the best since it will lead to cases randomly working and not working.